src/Microsoft.ServiceFabric.Data.Indexing.Persistent/ReliableIndexedDictionary.cs

```diff
@@ -403,14 +392,14 @@ public Task<IAsyncEnumerable<TFilter>> CreateIndexEnumerableAsync<TFilter>(ITran
 					continue;
 
 				// TODO: since we're doing snapshot reads, the value may have changed since we read the index.  We should validate the key-value still match the filter/search/etc.
-
+                // Note: In queryable this is still done because the OData are still applied to the remaining KeyValue set
 				results.Add(new KeyValuePair<TKey, TValue>(key, result.Value));
 			}
 
-			return results;
+			return Enumerable.AsEnumerable(results);
 		}
 
-		private FilterableIndex<TKey, TValue, TFilter> GetFilterableIndex<TFilter>(string indexName)
+		public FilterableIndex<TKey, TValue, TFilter> GetFilterableIndex<TFilter>(string indexName)
```

Comment:
Do these methods (GetFilterableIndex and GetAllAsync) need to be public?  Prefer private unless strictly necessary.

Reply:
For GetAllAsync, the scenario I have is 
Query: Name eq "Antonio" and Age eq 15, top=5
before that would have been FilterAsync("Antonio", 5)
but optimally I would like to 
Dictionary.GetAll(Intersect(FilterForKeys(Name), FilterForKeys(Age)))
It is optimal to merge the two Keysets and then retrieving from the original dictionary then retrieving and then merging. Which is why I made it public, loosened a count limit on some functions, and added the Key versiosn of Range/Filter. Let me know if this makes sense!

Reply:
Makes sense - just add everything you need publicly to the interface.